### PR TITLE
Plans 2023: data store slugs and more

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to #76024 

## Proposed Changes

Continuation of #76024 to introduce the plans data store into the 2023 plan grids. 
- Introduces more slugs to cover the rest of the plans
- `billingTerm` included in the plan product structure (used in follow up PR)
- a selector to fetch plan product given a respective slug (used in follow up PR)

### Author Notes

The draft P2 post ( dvytD-p2  ) contains details about this work and motivation. For the time being, we move enough content into the plans data store to surface it for usage in the plans 2023 grids (continuing on how it's already set up to work).

## Testing Instructions

There are no UI integrations. One way to test:
- go to `/plans/[SITE]`
- open Redux dev tools and inspect the latest image for `automattic/onboard/plans`
- under `State`, confirm all the variant `planProducts` for different periods are visible
- confirm `billingTerm` is included in that structure

Screenshot 2023-04-26 at 12 09 42 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/data-stores/src/plans/constants.ts [hunk 1]
```diff
@@ -61,3 +94,16 @@ export const PERIOD_LIST = [
...
+export const TERM_ANNUALLY = 'TERM_ANNUALLY';
+export const TERM_BIENNIALLY = 'TERM_BIENNIALLY';
+export const TERM_TRIENNIALLY = 'TERM_TRIENNIALLY';
+
+export const TERMS_LIST = [
+	TERM_MONTHLY,
+	TERM_ANNUALLY,
+	TERM_BIENNIALLY,
+	TERM_TRIENNIALLY,
+] as const;
```
Comment (chriskmnds) [inline packages/data-stores/src/plans/constants.ts:109]: These should probably move down the line to some other central/shared location. Presumably term constants can be shared by the [products store](https://github.com/Automattic/wp-calypso/tree/trunk/packages/data-stores/src/products-list), as we build out that part more. We have "shared-types" in the root, so "shared-something" might make sense.

File: packages/data-stores/src/plans/constants.ts [hunk 2]
```diff
@@ -42,7 +53,29 @@ export const monthlySlugs = [
...
+
+export const triennialSlugs = [
+	PLAN_PERSONAL_3_YEARS,
+	PLAN_PREMIUM_3_YEARS,
+	PLAN_BUSINESS_3_YEARS,
+	PLAN_ECOMMERCE_3_YEARS,
+] as const;
+
+export const plansProductSlugs = [
```
Comment (chriskmnds) [inline packages/data-stores/src/plans/constants.ts:72]: Needs the WooCommerce store slugs. Just adding any remaining slugs here should suffice - the resolver will normalize and populate the `PlanProduct` structure.

**Review by southp — APPROVED** [review]
Together with the real world use cases in https://github.com/Automattic/wp-calypso/pull/74797/files, the change makes sense to me 🚢

**Comment by chriskmnds** [comment]
We will not be going in this direction. The decision has been made already to publish `@automattic/calypso-products` to NPM and continue using metadata from there. In an ideal world, we'd be centralizing everything under said data store - however, that is a rather far-fetched target (discussion: pdvytD-bG-p2 ) given priorities and so much more to tackle with the NPM refactors. It's a good enough compromise.

Closing now as irrelevant.

**Pull request closed without merging**